main/survey/survey.lib.php

```diff
@@ -5189,4 +5323,62 @@ static function show_link_available($user_id, $survey_code, $user_answer)
             return false;
         }
     }
+
+    /**
+     * Display survey question chart
+     * @param	array	Chart data
+     * @param	boolean	Tells if the chart has a serie. False by default
+     * @return	void 	(direct output)
+     */
+    static function drawChart($chartData, $hasSerie = false)
+    {
+        $htmlChart = '<div id="chartContainer" class="span12">'.
+            api_get_js("d3/d3.v3.4.8.js").'
+            <script src="http://dimplejs.org/dist/dimple.v2.1.0.min.js"></script>
```

Comment:
Not a good idea to use external js like this. 
The lib should be added in Chamilo if it has a GPLv3 compatible license.